app/SectionApp/TodoContainer/Repositories/StepRepository.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\SectionApp\TodoContainer\Repositories;
+
+use App\SectionApp\TodoContainer\Models\Step;
+use App\SectionApp\TodoContainer\Repositories\Interfaces\StepRepositoryInterface;
+
+/**
+ * Class StepRepository
+ *
+ * This class is responsible for interacting with the step model.
+ */
+class StepRepository implements StepRepositoryInterface
+{
+    /**
+     * Create steps of todo.
+     *
+     * @param array<int, array<string, string|int>> $data
+     */
+    public function create(array $data): void
+    {
+        Step::insert($data);
+    }
+
+    /**
+     * Delete step by todo_id
+     */
+    public function delete(int $todoId): void
+    {
+        Step::where('todo_id', $todoId)->delete();
```

Comment:
User có thể truyền id ko tồn tại nên check Todo có tồn tại hay không, nếu không thì return 404
```suggestion
        Todo::findOrFail($todoId)->steps()->delete();
```